eng/common/scripts/Helpers/Resource-Helpers.ps1

```diff
@@ -7,6 +7,18 @@ function Get-PurgeableGroupResources {
   )
   $purgeableResources = @()
 
+  # Discover Managed HSMs first since they are a premium resource.
```

Comment:
It might not be worth pushing this change to all the other repos but if you want to do it locally until we get through the list that is fine but I leave it up to you.